src/aliceVision/feature/apriltag/ImageDescriber_APRILTAG.cpp

```diff
@@ -0,0 +1,117 @@
+// This file is part of the AliceVision project.
+// Copyright (c) 2016 AliceVision contributors.
+// Copyright (c) 2012 openMVG contributors.
+// This Source Code Form is subject to the terms of the Mozilla Public License,
+// v. 2.0. If a copy of the MPL was not distributed with this file,
+// You can obtain one at https://mozilla.org/MPL/2.0/.
+
+#include "ImageDescriber_APRILTAG.hpp"
+#include <aliceVision/gpu/gpu.hpp>
+#include <aliceVision/system/Logger.hpp>
+
+#include <apriltag/apriltag.h>
+#include <apriltag/tag16h5.h>
+#include <opencv2/core/core.hpp>
+
+namespace aliceVision {
+namespace feature {
+
+ImageDescriber_APRILTAG::AprilTagParameters::AprilTagParameters()
+{
+}
+
+void ImageDescriber_APRILTAG::AprilTagParameters::setPreset(EImageDescriberPreset preset)
+{
+}
+
+ImageDescriber_APRILTAG::ImageDescriber_APRILTAG()
+  : ImageDescriber()
+  {}
+
+void ImageDescriber_APRILTAG::allocate(std::unique_ptr<Regions> &regions) const
+{
+  regions.reset( new APRILTAG_Regions );
+}
+
+void ImageDescriber_APRILTAG::setConfigurationPreset(ConfigurationPreset preset)
+{
+  _params.setPreset(preset.descPreset);
+}
+
+EImageDescriberType ImageDescriber_APRILTAG::getDescriberType() const
+{
+  return EImageDescriberType::APRILTAG16H5;
+}
+
+bool ImageDescriber_APRILTAG::describe(const image::Image<unsigned char>& image,
+    std::unique_ptr<Regions> &regions,
+    const image::Image<unsigned char> * mask)
+{
+  allocate(regions);
+  ALICEVISION_LOG_DEBUG(" regions.get() = " << regions.get());
+  APRILTAG_Regions * regionsCasted = dynamic_cast<APRILTAG_Regions*>(regions.get());
+  ALICEVISION_LOG_DEBUG(" regionsCasted = " << regionsCasted);
+
+  apriltag_family_t *tf = tag16h5_create();
+  apriltag_detector_t *td = apriltag_detector_create();
+  apriltag_detector_add_family(td, tf);
+  td->quad_decimate = 2.0;
+  td->quad_sigma = 0.0; // blur
+  td->nthreads = 1;
+  td->debug = 0;
+  td->refine_edges = 1;
+
+  const cv::Mat graySrc(cv::Size(image.Width(), image.Height()), CV_8UC1, (unsigned char *) image.data(), cv::Mat::AUTO_STEP);
+  // Make an image_u8_t header for the Mat data
```

Comment:
I think aprilTag (the library) does not depends on opencv.
Opencv is optional in alicevision, it would be nice keep aprilTag independent from opencv.
Otherwise, even in the cmake u should tie aprilTag to the fact that opencv is ON
Here for example i think u can drop the cv::Mat and use `image` to fill the `image_u8_t`